### PR TITLE
Add `BLPOP` and `BRPOP` commands.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
* [`BLPOP`](https://redis.io/commands/blpop/)
* [`BRPOP`](https://redis.io/commands/brpop/)
* TODO: update changelog if needed

ref: https://github.com/aws/glide-for-redis/pull/1113

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 1]
```diff
@@ -225,4 +225,52 @@ public interface ListBaseCommands {
...
+     * Pops an element from the head of the first list that is non-empty, with the given keys being
+     * checked in the order that they are given.<br>
+     * Blocks the connection when there are no elements to pop from any of the given lists.
+     *
+     * @see <a href="https://redis.io/commands/blpop/">redis.io</a> for details.
+     * @apiNote <code>BLPOP</code> is a client blocking command, see <a
+     *     href="https://github.com/aws/glide-for-redis/wiki/General-Concepts#blocking-commands">Blocking
+     *     Commands</a> for more details and best practices.
+     * @param keys The <code>keys</code> of the lists to pop from.
+     * @param timeout The <code>timeout</code> of blocking in seconds; <code>0</code> causes to block
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:239]: nit: same suggestions for the other blpop and brpop docs
```suggestion
     * @param timeout The number of seconds to wait for a blocking <code>BLPOP</code> operation to complete. A value of <code>0</code> will block indefinitely.
```

File: java/client/src/main/java/glide/api/commands/ListBaseCommands.java [hunk 2]
```diff
@@ -225,4 +225,52 @@ public interface ListBaseCommands {
...
+     * }</pre>
+     */
+    CompletableFuture<String[]> blpop(String[] keys, double timeout);
+
+    /**
+     * Pops an element from the tail of the first list that is non-empty, with the given keys being
+     * checked in the order that they are given.<br>
+     * Blocks the connection when there are no elements to pop from any of the given lists.
+     *
+     * @see <a href="https://redis.io/commands/blpop/">redis.io</a> for details.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/ListBaseCommands.java:258]:
```suggestion
     * @see <a href="https://redis.io/commands/brpop/">redis.io</a> for details.
```

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 3]
```diff
@@ -1286,6 +1288,52 @@ public T type(@NonNull String key) {
...
+        protobufTransaction.addCommands(buildCommand(Blpop, commandArgs));
+        return getThis();
+    }
+
+    /**
+     * Pops an element from the tail of the first list that is non-empty, with the given keys being
+     * checked in the order that they are given.<br>
+     * Blocks the connection when there are no elements to pop from any of the given lists.
+     *
+     * @see <a href="https://redis.io/commands/blpop/">redis.io</a> for details.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1319]:
```suggestion
     * @see <a href="https://redis.io/commands/brpop/">redis.io</a> for details.
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 4]
```diff
@@ -1066,4 +1066,52 @@ public void type(BaseClient client) {
...
+        assertArrayEquals(new String[] {listKey1, value1}, response);
+
+        // nothing popped out
+        assertNull(client.brpop(new String[] {listKey2}, 0.001).get());
+
+        // Key exists, but it is not a list
+        assertEquals(OK, client.set("foo", "bar").get());
+        ExecutionException executionException =
+                assertThrows(
+                        ExecutionException.class, () -> client.blpop(new String[] {"foo"}, .0001).get());
```
Comment (aaron-congo) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1114]:
```suggestion
                        ExecutionException.class, () -> client.brpop(new String[] {"foo"}, .0001).get());
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 5]
```diff
@@ -1168,4 +1169,58 @@ public void type(BaseClient client) {
...
+        String value2 = "value2-" + UUID.randomUUID();
+        assertEquals(2, client.lpush(listKey1, new String[] {value1, value2}).get());
+
+        var response = client.blpop(new String[] {listKey1, listKey2}, 0.5).get();
+        assertArrayEquals(new String[] {listKey1, value2}, response);
+
+        // nothing popped out
+        assertNull(
+                client
+                        .blpop(new String[] {listKey2}, REDIS_VERSION.isLowerThan("7.0.0") ? 1. : 0.001)
```
Comment (SanHalacogluImproving) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1189]: What is that you are testing via the ternary operator?

**Review by SanHalacogluImproving — APPROVED** [review]

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 5 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1189]: Redis 6 does not support FP values. It rounds 0.0001 to 0 and blocks infinitely.

**Pull request merged**